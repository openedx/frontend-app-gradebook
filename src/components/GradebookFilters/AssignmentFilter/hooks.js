import {
  selectors,
  actions,
  thunkActions,
} from 'data/redux/hooks';

export const useAssignmentFilterData = ({
  updateQueryParams,
}) => {
  const assignmentFilterOptions = selectors.filters.useSelectableAssignmentLabels();
  const selectedAssignmentLabel = selectors.filters.useSelectedAssignmentLabel() || '';

  const updateAssignmentFilter = actions.filters.useUpdateAssignment();
  const conditionalFetch = thunkActions.grades.useFetchGradesIfAssignmentGradeFiltersSet();

  const handleChange = ({ target: { value: assignment } }) => {
    const selectedFilterOption = assignmentFilterOptions.find(
      ({ label }) => label === assignment,
    );
    const { type, id } = selectedFilterOption || {};
    updateAssignmentFilter({ label: assignment, type, id });
    updateQueryParams({ assignment: id });
    conditionalFetch();
  };

  return {
    handleChange,
    selectedAssignmentLabel,
    assignmentFilterOptions,
  };
};

export default useAssignmentFilterData;
