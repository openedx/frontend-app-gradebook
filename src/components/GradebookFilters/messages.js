import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  assignments: {
    id: 'gradebook.GradebookFilters.assignmentsFilterLabel',
    defaultMessage: 'Assignments',
    description: 'Assignment filter group label in Gradebook Filters',
  },
  overallGrade: {
    id: 'gradebook.GradebookFilters.overallGradeFilterLabel',
    defaultMessage: 'Overall Grade',
    description: 'Overall Grade filter group label in Gradebook Filters',
  },
  studentGroups: {
    id: 'gradebook.GradebookFilters.studentGroupsFilterLabel',
    defaultMessage: 'Student Groups',
    description: 'Student Groups filter group label in Gradebook Filters',
  },
  includeCourseTeamMembers: {
    id: 'gradebook.GradebookFilters.includeCourseTeamMembersFilterLabel',
    defaultMessage: 'Include Course Team Members',
    description: 'Include Course Team Members filter label in Gradebook Filters',
  },
  assignment: {
    id: 'gradebook.GradebookFilters.assignmentFilterLabel',
    defaultMessage: 'Assignment',
    description: 'Assignment filter select label in Gradebook Filters',
  },
  assignmentTypes: {
    id: 'gradebook.GradebookFilters.assignmentTypesLabel',
    defaultMessage: 'Assignment Types',
    description: 'Assignment Types filter select label in Gradebook Filters',
  },
  maxGrade: {
    id: 'gradebook.GradebookFilters.maxGradeFilterLabel',
    defaultMessage: 'Max Grade',
    description: 'Max-grade filter select label in Gradebook Filters',
  },
  minGrade: {
    id: 'gradebook.GradebookFilters.minGradeFilterLabel',
    defaultMessage: 'Min Grade',
    description: 'Min-grade filter select label in Gradebook Filters',
  },
  cohorts: {
    id: 'gradebook.GradebookFilters.cohorts',
    defaultMessage: 'Cohorts',
    description: 'Cohorts filter select label in Gradebook Filters',
  },
  cohortAll: {
    id: 'gradebook.GradebookFilters.cohortsAll',
    defaultMessage: 'Cohort-All',
    description: 'Cohorts filter select default in Gradebook Filters',
  },
  tracks: {
    id: 'gradebook.GradebookFilters.tracks',
    defaultMessage: 'Tracks',
    description: 'Tracks filter select label in Gradebook Filters',
  },
  trackAll: {
    id: 'gradebook.GradebookFilters.trackAll',
    defaultMessage: 'Track-All',
    description: 'Tracks filter select default in Gradebook Filters',
  },
  closeFilters: {
    id: 'gradebook.GradebookFilters.closeFilters',
    defaultMessage: 'Close Filters',
    description: 'Button label for Close button in Gradebook Filters',
  },
});

export default messages;
