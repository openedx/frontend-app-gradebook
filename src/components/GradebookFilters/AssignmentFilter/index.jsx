/* eslint-disable react/sort-comp, react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { FormattedMessage } from '@edx/frontend-platform/i18n';

import selectors from 'data/selectors';
import actions from 'data/actions';
import thunkActions from 'data/thunkActions';

import messages from '../messages';
import SelectGroup from '../SelectGroup';

const { fetchGradesIfAssignmentGradeFiltersSet } = thunkActions.grades;

export class AssignmentFilter extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    const assignment = event.target.value;
    const selectedFilterOption = this.props.assignmentFilterOptions.find(
      ({ label }) => label === assignment,
    );
    const { type, id } = selectedFilterOption || {};
    const typedValue = { label: assignment, type, id };
    this.props.updateAssignmentFilter(typedValue);
    this.props.updateQueryParams({ assignment: id });
    this.props.fetchGradesIfAssignmentGradeFiltersSet();
  }

  get options() {
    const mapper = ({ label, subsectionLabel }) => (
      <option key={label} value={label}>
        {label}: {subsectionLabel}
      </option>
    );
    return ([
      <option key="0" value="">All</option>,
      ...this.props.assignmentFilterOptions.map(mapper),
    ]);
  }

  render() {
    return (
      <div className="student-filters">
        <SelectGroup
          id="assignment"
          label={<FormattedMessage {...messages.assignment} />}
          value={this.props.selectedAssignment}
          onChange={this.handleChange}
          disabled={this.props.assignmentFilterOptions.length === 0}
          options={this.options}
        />
      </div>
    );
  }
}

AssignmentFilter.defaultProps = {
  assignmentFilterOptions: [],
  selectedAssignment: '',
};

AssignmentFilter.propTypes = {
  updateQueryParams: PropTypes.func.isRequired,
  // redux
  assignmentFilterOptions: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string,
    subsectionLabel: PropTypes.string,
    type: PropTypes.string,
    id: PropTypes.string,
  })),
  selectedAssignment: PropTypes.string,
  fetchGradesIfAssignmentGradeFiltersSet: PropTypes.func.isRequired,
  updateAssignmentFilter: PropTypes.func.isRequired,
};

export const mapStateToProps = (state) => {
  const { filters } = selectors;
  return {
    assignmentFilterOptions: filters.selectableAssignmentLabels(state),
    selectedAssignment: filters.selectedAssignmentLabel(state),
    selectedAssignmentType: filters.assignmentType(state),
    selectedCohort: filters.cohort(state),
    selectedTrack: filters.track(state),
  };
};

export const mapDispatchToProps = {
  updateAssignmentFilter: actions.filters.update.assignment,
  fetchGradesIfAssignmentGradeFiltersSet,
};

export default connect(mapStateToProps, mapDispatchToProps)(AssignmentFilter);
