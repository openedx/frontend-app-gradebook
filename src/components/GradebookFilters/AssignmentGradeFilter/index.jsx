/* eslint-disable react/sort-comp, react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { FormattedMessage } from '@edx/frontend-platform/i18n';
import { Button } from '@edx/paragon';

import selectors from 'data/selectors';
import actions from 'data/actions';
import thunkActions from 'data/thunkActions';

import messages from '../messages';
import PercentGroup from '../PercentGroup';

export class AssignmentGradeFilter extends React.Component {
  constructor(props) {
    super(props);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleSetMax = this.handleSetMax.bind(this);
    this.handleSetMin = this.handleSetMin.bind(this);
  }

  handleSubmit() {
    this.props.updateAssignmentLimits(this.props.localAssignmentLimits);
    this.props.fetchGrades();
    this.props.updateQueryParams(this.props.localAssignmentLimits);
  }

  handleSetMax({ target: { value } }) {
    this.props.setFilter({ assignmentGradeMax: value });
  }

  handleSetMin({ target: { value } }) {
    this.props.setFilter({ assignmentGradeMin: value });
  }

  render() {
    const {
      localAssignmentLimits: { assignmentGradeMax, assignmentGradeMin },
    } = this.props;
    return (
      <div className="grade-filter-inputs">
        <PercentGroup
          id="assignmentGradeMin"
          label={<FormattedMessage {...messages.minGrade} />}
          value={assignmentGradeMin}
          disabled={!this.props.selectedAssignment}
          onChange={this.handleSetMin}
        />
        <PercentGroup
          id="assignmentGradeMax"
          label={<FormattedMessage {...messages.maxGrade} />}
          value={assignmentGradeMax}
          disabled={!this.props.selectedAssignment}
          onChange={this.handleSetMax}
        />
        <div className="grade-filter-action">
          <Button
            type="submit"
            variant="outline-secondary"
            name="assignmentGradeMinMax"
            disabled={!this.props.selectedAssignment}
            onClick={this.handleSubmit}
          >
            Apply
          </Button>
        </div>
      </div>
    );
  }
}

AssignmentGradeFilter.defaultProps = {
  selectedAssignment: '',
};

AssignmentGradeFilter.propTypes = {
  updateQueryParams: PropTypes.func.isRequired,

  // redux
  fetchGrades: PropTypes.func.isRequired,
  localAssignmentLimits: PropTypes.shape({
    assignmentGradeMax: PropTypes.string,
    assignmentGradeMin: PropTypes.string,
  }).isRequired,
  selectedAssignment: PropTypes.string,
  setFilter: PropTypes.func.isRequired,
  updateAssignmentLimits: PropTypes.func.isRequired,
};

export const mapStateToProps = (state) => ({
  localAssignmentLimits: selectors.app.assignmentGradeLimits(state),
  selectedAssignment: selectors.filters.selectedAssignmentLabel(state),
});

export const mapDispatchToProps = {
  fetchGrades: thunkActions.grades.fetchGrades,
  setFilter: actions.app.setLocalFilter,
  updateAssignmentLimits: actions.filters.update.assignmentLimits,
};

export default connect(mapStateToProps, mapDispatchToProps)(AssignmentGradeFilter);
