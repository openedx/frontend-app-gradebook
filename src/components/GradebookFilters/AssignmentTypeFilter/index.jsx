/* eslint-disable react/sort-comp, react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { FormattedMessage } from '@edx/frontend-platform/i18n';

import selectors from 'data/selectors';
import actions from 'data/actions';

import SelectGroup from '../SelectGroup';
import messages from '../messages';

export class AssignmentTypeFilter extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    const assignmentType = event.target.value;
    this.props.filterAssignmentType(assignmentType);
    this.props.updateQueryParams({ assignmentType });
  }

  get options() {
    const mapper = (entry) => (
      <option key={entry} value={entry}>{entry}</option>
    );
    return [
      <option key="0" value="">All</option>,
      ...this.props.assignmentTypes.map(mapper),
    ];
  }

  render() {
    return (
      <div className="student-filters">
        <SelectGroup
          id="assignment-types"
          label={<FormattedMessage {...messages.assignmentTypes} />}
          value={this.props.selectedAssignmentType}
          onChange={this.handleChange}
          disabled={this.props.assignmentFilterOptions.length === 0}
          options={this.options}
        />
      </div>
    );
  }
}

AssignmentTypeFilter.defaultProps = {
  assignmentTypes: [],
  assignmentFilterOptions: [],
  selectedAssignmentType: '',
};

AssignmentTypeFilter.propTypes = {
  updateQueryParams: PropTypes.func.isRequired,

  // redux
  assignmentTypes: PropTypes.arrayOf(PropTypes.string),
  assignmentFilterOptions: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string,
    subsectionLabel: PropTypes.string,
  })),
  filterAssignmentType: PropTypes.func.isRequired,
  selectedAssignmentType: PropTypes.string,
};

export const mapStateToProps = (state) => ({
  assignmentTypes: selectors.assignmentTypes.allAssignmentTypes(state),
  assignmentFilterOptions: selectors.filters.selectableAssignmentLabels(state),
  selectedAssignmentType: selectors.filters.assignmentType(state),
});

export const mapDispatchToProps = {
  filterAssignmentType: actions.filters.update.assignmentType,
};

export default connect(mapStateToProps, mapDispatchToProps)(AssignmentTypeFilter);
