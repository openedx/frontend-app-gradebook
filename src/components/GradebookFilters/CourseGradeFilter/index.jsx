/* eslint-disable react/sort-comp, react/button-has-type, import/no-named-as-default */
import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { Button } from '@edx/paragon';
import { FormattedMessage } from '@edx/frontend-platform/i18n';

import selectors from 'data/selectors';
import actions from 'data/actions';
import thunkActions from 'data/thunkActions';

import messages from '../messages';
import PercentGroup from '../PercentGroup';

export class CourseGradeFilter extends React.Component {
  constructor(props) {
    super(props);
    this.handleApplyClick = this.handleApplyClick.bind(this);
    this.handleUpdateMin = this.handleUpdateMin.bind(this);
    this.handleUpdateMax = this.handleUpdateMax.bind(this);
    this.updateCourseGradeFilters = this.updateCourseGradeFilters.bind(this);
  }

  handleApplyClick() {
    if (this.props.areLimitsValid) {
      this.updateCourseGradeFilters();
    }
  }

  handleUpdateMin({ target: { value } }) {
    this.props.setLocalFilter({ courseGradeMin: value });
  }

  handleUpdateMax({ target: { value } }) {
    this.props.setLocalFilter({ courseGradeMax: value });
  }

  updateCourseGradeFilters() {
    this.props.updateFilter(this.props.localCourseLimits);
    this.props.fetchGrades();
    this.props.updateQueryParams(this.props.localCourseLimits);
  }

  render() {
    const {
      localCourseLimits: { courseGradeMin, courseGradeMax },
    } = this.props;
    return (
      <>
        <div className="grade-filter-inputs">
          <PercentGroup
            id="minimum-grade"
            label={<FormattedMessage {...messages.minGrade} />}
            value={courseGradeMin}
            onChange={this.handleUpdateMin}
          />
          <PercentGroup
            id="maximum-grade"
            label={<FormattedMessage {...messages.maxGrade} />}
            value={courseGradeMax}
            onChange={this.handleUpdateMax}
          />
        </div>
        <div className="grade-filter-action">
          <Button
            variant="outline-secondary"
            onClick={this.handleApplyClick}
          >
            Apply
          </Button>
        </div>
      </>
    );
  }
}

CourseGradeFilter.propTypes = {
  updateQueryParams: PropTypes.func.isRequired,

  // Redux
  areLimitsValid: PropTypes.bool.isRequired,
  fetchGrades: PropTypes.func.isRequired,
  localCourseLimits: PropTypes.shape({
    courseGradeMin: PropTypes.string.isRequired,
    courseGradeMax: PropTypes.string.isRequired,
  }).isRequired,
  setLocalFilter: PropTypes.func.isRequired,
  updateFilter: PropTypes.func.isRequired,
};

export const mapStateToProps = (state) => ({
  areLimitsValid: selectors.app.areCourseGradeFiltersValid(state),
  localCourseLimits: selectors.app.courseGradeLimits(state),
});

export const mapDispatchToProps = {
  fetchGrades: thunkActions.grades.fetchGrades,
  setLocalFilter: actions.app.setLocalFilter,
  updateFilter: actions.filters.update.courseGradeLimits,
};

export default connect(mapStateToProps, mapDispatchToProps)(CourseGradeFilter);
