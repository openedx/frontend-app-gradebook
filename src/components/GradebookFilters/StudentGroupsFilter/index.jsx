/* eslint-disable react/sort-comp, react/button-has-type, import/no-named-as-default */
import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';

import actions from 'data/actions';
import selectors from 'data/selectors';
import thunkActions from 'data/thunkActions';

import messages from '../messages';
import SelectGroup from '../SelectGroup';

export const optionFactory = ({ data, defaultOption, key }) => [
  <option value={defaultOption} key="0">{defaultOption}</option>,
  ...data.map(
    entry => (<option key={entry[key]} value={entry.name}>{entry.name}</option>),
  ),
];

export class StudentGroupsFilter extends React.Component {
  constructor(props) {
    super(props);
    this.mapCohortsEntries = this.mapCohortsEntries.bind(this);
    this.mapTracksEntries = this.mapTracksEntries.bind(this);
    this.updateCohorts = this.updateCohorts.bind(this);
    this.updateTracks = this.updateTracks.bind(this);
  }

  mapCohortsEntries() {
    return optionFactory({
      data: this.props.cohorts,
      defaultOption: this.translate(messages.cohortAll),
      key: 'id',
    });
  }

  mapTracksEntries() {
    return optionFactory({
      data: this.props.tracks,
      defaultOption: this.translate(messages.trackAll),
      key: 'slug',
    });
  }

  selectedTrackSlugFromEvent({ target: { value } }) {
    const selectedTrackItem = this.props.tracksByName[value];
    return selectedTrackItem ? selectedTrackItem.slug : null;
  }

  selectedCohortIdFromEvent({ target: { value } }) {
    const selectedCohortItem = this.props.cohortsByName[value];
    return selectedCohortItem ? selectedCohortItem.id.toString() : null;
  }

  updateTracks(event) {
    const track = this.selectedTrackSlugFromEvent(event);
    this.props.updateQueryParams({ track });
    this.props.updateTrack(track);
    this.props.fetchGrades();
  }

  updateCohorts(event) {
    const cohort = this.selectedCohortIdFromEvent(event);
    this.props.updateQueryParams({ cohort });
    this.props.updateCohort(cohort);
    this.props.fetchGrades();
  }

  translate(message) {
    return this.props.intl.formatMessage(message);
  }

  render() {
    return (
      <>
        <SelectGroup
          id="Tracks"
          label={this.translate(messages.tracks)}
          value={this.props.selectedTrackEntry.name}
          onChange={this.updateTracks}
          options={this.mapTracksEntries()}
        />
        <SelectGroup
          id="Cohorts"
          label={this.translate(messages.cohorts)}
          value={this.props.selectedCohortEntry.name}
          disabled={this.props.cohorts.length === 0}
          onChange={this.updateCohorts}
          options={this.mapCohortsEntries()}
        />
      </>
    );
  }
}

StudentGroupsFilter.defaultProps = {
  cohorts: [],
  cohortsByName: {},
  selectedCohortEntry: { name: '' },
  selectedTrackEntry: { name: '' },
  tracks: [],
  tracksByName: {},
};

StudentGroupsFilter.propTypes = {
  updateQueryParams: PropTypes.func.isRequired,

  // injected
  intl: intlShape.isRequired,

  // redux
  cohorts: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    id: PropTypes.number,
  })),
  cohortsByName: PropTypes.objectOf(PropTypes.shape({
    name: PropTypes.string,
    id: PropTypes.number,
  })),
  fetchGrades: PropTypes.func.isRequired,
  selectedTrackEntry: PropTypes.shape({ name: PropTypes.string }),
  selectedCohortEntry: PropTypes.shape({ name: PropTypes.string }),
  tracks: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    slug: PropTypes.string,
  })),
  tracksByName: PropTypes.objectOf(PropTypes.shape({
    name: PropTypes.string,
    slug: PropTypes.string,
  })),
  updateCohort: PropTypes.func.isRequired,
  updateTrack: PropTypes.func.isRequired,
};

export const mapStateToProps = (state) => ({
  cohorts: selectors.cohorts.allCohorts(state),
  cohortsByName: selectors.cohorts.cohortsByName(state),
  selectedCohortEntry: selectors.root.selectedCohortEntry(state),
  selectedTrackEntry: selectors.root.selectedTrackEntry(state),
  tracks: selectors.tracks.allTracks(state),
  tracksByName: selectors.tracks.tracksByName(state),
});

export const mapDispatchToProps = {
  fetchGrades: thunkActions.grades.fetchGrades,
  updateCohort: actions.filters.update.cohort,
  updateTrack: actions.filters.update.track,
};

export default injectIntl(connect(mapStateToProps, mapDispatchToProps)(StudentGroupsFilter));
