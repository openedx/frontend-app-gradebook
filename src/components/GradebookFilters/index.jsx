/* eslint-disable react/sort-comp, import/no-named-as-default */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import {
  Collapsible,
  Icon,
  IconButton,
  Form,
} from '@edx/paragon';
import { Close } from '@edx/paragon/icons';
import { FormattedMessage, injectIntl, intlShape } from '@edx/frontend-platform/i18n';

import actions from 'data/actions';
import selectors from 'data/selectors';
import thunkActions from 'data/thunkActions';

import messages from './messages';
import AssignmentTypeFilter from './AssignmentTypeFilter';
import AssignmentFilter from './AssignmentFilter';
import AssignmentGradeFilter from './AssignmentGradeFilter';
import CourseGradeFilter from './CourseGradeFilter';
import StudentGroupsFilter from './StudentGroupsFilter';

export class GradebookFilters extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      includeCourseRoleMembers: this.props.includeCourseRoleMembers,
    };
    this.handleIncludeTeamMembersChange = this.handleIncludeTeamMembersChange.bind(this);
  }

  handleIncludeTeamMembersChange(event) {
    const includeCourseRoleMembers = event.target.checked;
    this.setState({ includeCourseRoleMembers });
    this.props.updateIncludeCourseRoleMembers(includeCourseRoleMembers);
    this.props.fetchGrades();
    this.props.updateQueryParams({ includeCourseRoleMembers });
  }

  collapsibleGroup = (title, content) => (
    <Collapsible
      title={<FormattedMessage {...title} />}
      defaultOpen
      className="filter-group mb-3"
    >
      {content}
    </Collapsible>
  );

  render() {
    const {
      intl,
      updateQueryParams,
    } = this.props;
    return (
      <>
        <div className="filter-sidebar-header">
          <h2><Icon className="fa fa-filter" /></h2>
          <IconButton
            className="p-1"
            onClick={this.props.closeMenu}
            iconAs={Icon}
            src={Close}
            alt={intl.formatMessage(messages.closeFilters)}
            aria-label={intl.formatMessage(messages.closeFilters)}
          />
        </div>

        {this.collapsibleGroup(messages.assignments, (
          <div>
            <AssignmentTypeFilter updateQueryParams={updateQueryParams} />
            <AssignmentFilter updateQueryParams={updateQueryParams} />
            <AssignmentGradeFilter updateQueryParams={updateQueryParams} />
          </div>
        ))}

        {this.collapsibleGroup(messages.overallGrade, (
          <CourseGradeFilter updateQueryParams={updateQueryParams} />
        ))}

        {this.collapsibleGroup(messages.studentGroups, (
          <StudentGroupsFilter updateQueryParams={updateQueryParams} />
        ))}

        {this.collapsibleGroup(messages.includeCourseTeamMembers, (
          <Form.Checkbox
            checked={this.state.includeCourseRoleMembers}
            onChange={this.handleIncludeTeamMembersChange}
          >
            <FormattedMessage {...messages.includeCourseTeamMembers} />
          </Form.Checkbox>
        ))}
      </>
    );
  }
}
GradebookFilters.defaultProps = {
  includeCourseRoleMembers: false,
};
GradebookFilters.propTypes = {
  updateQueryParams: PropTypes.func.isRequired,
  // injected
  intl: intlShape.isRequired,
  // redux
  closeMenu: PropTypes.func.isRequired,
  fetchGrades: PropTypes.func.isRequired,
  includeCourseRoleMembers: PropTypes.bool,
  updateIncludeCourseRoleMembers: PropTypes.func.isRequired,
};

export const mapStateToProps = (state) => ({
  includeCourseRoleMembers: selectors.filters.includeCourseRoleMembers(state),
});

export const mapDispatchToProps = {
  closeMenu: thunkActions.app.filterMenu.close,
  fetchGrades: thunkActions.grades.fetchGrades,
  updateIncludeCourseRoleMembers: actions.filters.update.includeCourseRoleMembers,
};

export default injectIntl(connect(mapStateToProps, mapDispatchToProps)(GradebookFilters));
