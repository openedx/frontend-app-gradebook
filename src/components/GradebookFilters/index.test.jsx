import React from 'react';
import { shallow } from 'enzyme';

import { useIntl } from '@edx/frontend-platform/i18n';
import { Collapsible } from '@edx/paragon';

import { formatMessage } from 'testUtils';

import AssignmentTypeFilter from './AssignmentTypeFilter';
import AssignmentFilter from './AssignmentFilter';
import AssignmentGradeFilter from './AssignmentGradeFilter';
import CourseGradeFilter from './CourseGradeFilter';
import StudentGroupsFilter from './StudentGroupsFilter';
import messages from './messages';

import useGradebookFiltersData from './hooks';
import GradebookFilters from '.';

jest.mock('./AssignmentTypeFilter', () => 'AssignmentTypeFilter');
jest.mock('./AssignmentFilter', () => 'AssignmentFilter');
jest.mock('./AssignmentGradeFilter', () => 'AssignmentGradeFilter');
jest.mock('./CourseGradeFilter', () => 'CourseGradeFilter');
jest.mock('./StudentGroupsFilter', () => 'StudentGroupsFilter');

jest.mock('./hooks', () => ({ __esModule: true, default: jest.fn() }));

const hookProps = {
  closeMenu: jest.fn().mockName('hook.closeMenu'),
  includeCourseTeamMembers: {
    value: true,
    handleChange: jest.fn().mockName('hook.handleChange'),
  },
};
useGradebookFiltersData.mockReturnValue(hookProps);

let el;
const updateQueryParams = jest.fn();

describe('GradebookFilters', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    el = shallow(<GradebookFilters updateQueryParams={updateQueryParams} />);
  });
  describe('behavior', () => {
    it('initializes hooks', () => {
      expect(useGradebookFiltersData).toHaveBeenCalledWith({ updateQueryParams });
      expect(useIntl).toHaveBeenCalledWith();
    });
  });
  describe('render', () => {
    test('snapshot', () => {
      expect(el).toMatchSnapshot();
    });
    test('Assignment filters', () => {
      expect(el.find(Collapsible).at(0).children()).toMatchObject(shallow(
        <div>
          <AssignmentTypeFilter updateQueryParams={updateQueryParams} />
          <AssignmentFilter updateQueryParams={updateQueryParams} />
          <AssignmentGradeFilter updateQueryParams={updateQueryParams} />
        </div>,
      ));
    });
    test('CourseGrade filters', () => {
      expect(el.find(Collapsible).at(1).children()).toMatchObject(shallow(
        <CourseGradeFilter updateQueryParams={updateQueryParams} />,
      ));
    });
    test('StudentGroups filters', () => {
      expect(el.find(Collapsible).at(2).children()).toMatchObject(shallow(
        <StudentGroupsFilter updateQueryParams={updateQueryParams} />,
      ));
    });
    test('includeCourseTeamMembers', () => {
      const checkbox = el.find(Collapsible).at(3).children();
      expect(checkbox.props()).toEqual({
        checked: true,
        onChange: hookProps.includeCourseTeamMembers.handleChange,
        children: formatMessage(messages.includeCourseTeamMembers),
      });
    });
  });
});
