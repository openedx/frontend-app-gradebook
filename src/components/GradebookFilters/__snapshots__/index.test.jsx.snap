// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GradebookFilters render snapshot 1`] = `
<Fragment>
  <div
    className="filter-sidebar-header"
  >
    <h2>
      <Icon
        className="fa fa-filter"
      />
    </h2>
    <IconButton
      alt="Close Filters"
      aria-label="Close Filters"
      className="p-1"
      iconAs="Icon"
      onClick={[MockFunction hook.closeMenu]}
      src="Close"
    />
  </div>
  <Collapsible
    className="filter-group mb-3"
    defaultOpen={true}
    title="Assignments"
  >
    <div>
      <AssignmentTypeFilter
        updateQueryParams={[MockFunction]}
      />
      <AssignmentFilter
        updateQueryParams={[MockFunction]}
      />
      <AssignmentGradeFilter
        updateQueryParams={[MockFunction]}
      />
    </div>
  </Collapsible>
  <Collapsible
    className="filter-group mb-3"
    defaultOpen={true}
    title="Overall Grade"
  >
    <CourseGradeFilter
      updateQueryParams={[MockFunction]}
    />
  </Collapsible>
  <Collapsible
    className="filter-group mb-3"
    defaultOpen={true}
    title="Student Groups"
  >
    <StudentGroupsFilter
      updateQueryParams={[MockFunction]}
    />
  </Collapsible>
  <Collapsible
    className="filter-group mb-3"
    defaultOpen={true}
    title="Include Course Team Members"
  >
    <Form.Checkbox
      checked={true}
      onChange={[MockFunction hook.handleChange]}
    >
      Include Course Team Members
    </Form.Checkbox>
  </Collapsible>
</Fragment>
`;
